binaries/daemon/src/lib.rs

```diff
@@ -861,6 +880,80 @@ impl Daemon {
         Ok(status)
     }
 
+    async fn collect_and_send_metrics(&mut self) -> eyre::Result<()> {
+        use dora_message::daemon_to_coordinator::NodeMetrics;
+        use sysinfo::{Pid, ProcessRefreshKind, ProcessesToUpdate, System};
+
+        if self.coordinator_connection.is_none() {
+            return Ok(());
+        }
+
+        // Create system instance for metrics collection
+        let mut system = System::new();
+
+        // Collect metrics for all running dataflows
+        for (dataflow_id, dataflow) in &self.running {
+            let mut metrics = BTreeMap::new();
+
+            // Collect all PIDs for this dataflow
+            let pids: Vec<Pid> = dataflow
+                .running_nodes
+                .values()
+                .filter_map(|node| node.pid.map(Pid::from_u32))
+                .collect();
+
+            if !pids.is_empty() {
+                // Refresh all processes at once
+                system.refresh_processes_specifics(
+                    ProcessesToUpdate::Some(&pids),
+                    true,
+                    ProcessRefreshKind::everything(),
+                );
+
+                // Collect metrics for each node
+                for (node_id, running_node) in &dataflow.running_nodes {
+                    if let Some(pid) = running_node.pid {
+                        let sys_pid = Pid::from_u32(pid);
+                        if let Some(process) = system.process(sys_pid) {
+                            let disk_usage = process.disk_usage();
+                            metrics.insert(
+                                node_id.clone(),
+                                NodeMetrics {
+                                    pid,
+                                    cpu_usage: process.cpu_usage(),
+                                    memory_bytes: process.memory(),
+                                    disk_read_bytes: Some(disk_usage.read_bytes),
+                                    disk_write_bytes: Some(disk_usage.written_bytes),
```

Comment:
These numbers are since the last refresh, right? We should probably divide them by `metrics_interval` to get the avg numbers per second.